runtime/src/main/java/org/corfudb/runtime/LogReplicationClient.java

```diff
@@ -0,0 +1,277 @@
+        if (Strings.isNullOrEmpty(logicalGroup)) {
+            log.error("Invalid logicalGroup");
+            return false;
+        }
+        try {
+            return IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientInfoKey clientInfoKey = ClientInfoKey.newBuilder()
+                            .setClientName(clientName)
+                            .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                            .setGroupName(logicalGroup)
+                            .build();
+
+                    SinksInfoVal clientDestinations;
+
+                    boolean hasNullOrEmptyOrDuplicates =
+                            remoteDestinations.stream().anyMatch(Objects::isNull) ||
+                            remoteDestinations.stream().anyMatch(String::isEmpty) ||
+                            remoteDestinations.stream().distinct().count() < remoteDestinations.size();
+
+                    if (hasNullOrEmptyOrDuplicates) {
+                        log.error("Invalid destination(s) found in list");
+                        return false;
+                    }
+
+                    if (txn.isExists(LR_SOURCE_METADATA_TABLE_NAME, clientInfoKey)) {
+                        clientDestinations = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+
+                        for (String dest : remoteDestinations) {
+                            if (clientDestinations.getDestinationIdsList().contains(dest)) {
+                                return false;
+                            }
+                        }
+
+                        clientDestinations = clientDestinations.toBuilder().addAllDestinationIds(remoteDestinations).build();
+                    }
+                    else {
+                        clientDestinations = SinksInfoVal.newBuilder().addAllDestinationIds(remoteDestinations).build();
+                    }
+
+                    txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinations, null);
+                    txn.commit();
+
+                    return true;
+                }
+                catch (Exception e) {
+                    log.error("Unable to add destinations", e);
+                    return false;
+                }
+            }).setOptions(x -> x.setMaxRetryThreshold(Duration.ofSeconds(SYNC_THRESHOLD))).run();
+        } catch (InterruptedException e) {
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Remove destination from LogReplicationSourceMetadataTable
+     *
+     * @param logicalGroup String representative of the group of tables which are replicated
+     * @param destination Cluster from which the logicalGroup should be removed from further replication
+     * @return true if destination was removed
+     */
+    public boolean removeDestination(String logicalGroup, String destination) {
```

Comment:
Should we consider having APIs to also remove the logical group and to remove a list<destinations> ?